File: BioPerl/parse_pdb.pl
# Benchmark the parsing of a PDB file given as an argument

use Bio::Structure::IO;
use Time::HiRes qw(time);
use strict;

my $pdb_filepath = $ARGV[0];

sub parse {
    my $structio = Bio::Structure::IO->new(-file => $pdb_filepath);
    return $structio->next_structure;
}

my $start = time();
parse();
my $end = time();

print $end - $start, "\n";

File: Biopython/parse_pdb.py
# Benchmark the parsing of a PDB file given as an argument

import sys
import time
from Bio.PDB import PDBParser

pdb_filepath = sys.argv[1]
parser = PDBParser()

start = time.time()
parser.get_structure("", pdb_filepath)
end = time.time()

print(end - start)
